components/apimgt/org.wso2.carbon.apimgt.usage/org.wso2.carbon.apimgt.usage.client/src/main/java/org/wso2/carbon/apimgt/usage/client/DASRestClient.java

```diff
@@ -0,0 +1,256 @@
+/*
+* Copyright (c) 2015, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+* WSO2 Inc. licenses this file to you under the Apache License,
+* Version 2.0 (the "License"); you may not use this file except
+* in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing,
+* software distributed under the License is distributed on an
+* "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+* KIND, either express or implied. See the License for the
+* specific language governing permissions and limitations
+* under the License.
+*
+*/
+package org.wso2.carbon.apimgt.usage.client;
+
+import com.google.gson.Gson;
+import com.google.gson.JsonSyntaxException;
+import com.google.gson.reflect.TypeToken;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.http.HttpResponse;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.client.protocol.HttpClientContext;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.protocol.HttpContext;
+import org.wso2.carbon.apimgt.usage.client.bean.FirstAccessRequestSearchBean;
+import org.wso2.carbon.apimgt.usage.client.bean.Result;
+import org.wso2.carbon.apimgt.usage.client.bean.SearchRequestBean;
+import org.wso2.carbon.apimgt.usage.client.bean.TableExistResponseBean;
+import org.wso2.carbon.apimgt.usage.client.util.RestClientUtil;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.lang.reflect.Type;
+import java.util.List;
+
+/**
+ * This class contain the actual Client side implementation for the DAS rest client.
+ */
+public class DASRestClient {
+    private CloseableHttpClient httpClient;
+    private String dasUrl;
+    private String user;
+    private String pass;
+    private final Gson gson = new Gson();
+    private static final Log log = LogFactory.getLog(DASRestClient.class);
+
+    /**
+     * get instance providing DAS configuration
+     *
+     * @param url  DAS rest api location
+     * @param user DAS rest api username
+     * @param pass DAs rest api password
+     */
+    public DASRestClient(String url, String user, String pass) {
+        PoolingHttpClientConnectionManager cm = new PoolingHttpClientConnectionManager();
+        httpClient = HttpClients.custom().setConnectionManager(cm).build();
+        this.dasUrl = url;
+        this.user = user;
+        this.pass = pass;
+
```

Comment:
fixed in commit: https://github.com/wso2/carbon-apimgt/commit/c5071cbc7f3c39fed2489c288179e0fe93507544